moxy-compiler/src/test/java/moxy/compiler/NestedPresenterViewStateProviderTest.java

```diff
@@ -0,0 +1,54 @@
+package moxy.compiler;
+
+import com.google.testing.compile.Compilation;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.tools.JavaFileObject;
+
+import moxy.MvpProcessor;
+
+import static com.google.testing.compile.CompilationSubject.assertThat;
+
+@RunWith(Parameterized.class)
+public class NestedPresenterViewStateProviderTest extends CompilerTest {
+
+    @Parameterized.Parameter
+    public TestParams testParams;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static TestParams[] data() {
+        return new TestParams[] {
+            new TestParams(
+                "presenter.PresenterWrapper",
+                "presenter.PresenterWrapper$EmptyViewPresenter" + MvpProcessor.VIEW_STATE_PROVIDER_SUFFIX
+            ),
+        };
+    }
+
+    @Test
+    public void test() throws Exception {
+        JavaFileObject presenter = getSourceFile(testParams.sourceFileName);
+        JavaFileObject exceptedViewStateProvider = getSourceFile(testParams.compiledFileName);
```

Comment:
you probably meant "expected", right?

Reply:
Yes, right :) I took code from `ViewStateProviderTest` which has these mistakes and changed it for nested presenters.
This typo got my attention, I realized that "expected" and not "excepted" was meant, but I thought that we should maybe fix this later in a different branch. What if we merge this PR and after that I create a separate branch where I refactor "excepted" => "expected" (btw I already created the branch in which I replaced every occurrence of "excepted" to "expected" and everything works fine)?

Reply:
Okay, let's do it in separate requests.